internal/pkg/api/handleCheckin.go

```diff
@@ -251,13 +252,19 @@ func (ct *CheckinT) validateRequest(zlog zerolog.Logger, w http.ResponseWriter,
 		return val, err
 	}
 
+	rawRollbacks, err := parseAvailableRollbacks(zlog, agent, &req)
+	if err != nil {
+		return val, err
```

Comment:
This makes an invalid available_rollbacks field fail the whole checkin with a 4xx doesn't it?

I don't think having an invalid available_rollback field should prevent agent from reporting status or more importantly from getting an upgrade action to fix the bug causing corrupted available_rollbacks or a policy change to disable the rollback feature remotely. We ran into this with the download rate field, see https://github.com/elastic/fleet-server/issues/5164 for part of the series of issues that caused.

Can we strip any invalid available_rollbacks and proceed as normal, logging the error in fleet-server instead?

Reply:
It looks like the only error right now is if the parsed rollbacks can't be marshalled to JSON in which case we should behave as if available rollbacks weren't reported at all.

```go
		marshalled, err := json.Marshal(reqRollbacks)
		if err != nil {
			return nil, fmt.Errorf("marshalling available rollbacks: %w", err)
		}
```

Having some indication on agent would be useful, maybe returning an HTTP 202 could be one way to indicate something is up, but we'd need to make sure agents tolerate that properly. We don't need to do that though, I am happy as long as formatting issues don't prevent agents from using central management.

Reply:
I agree that an error parsing available rollbacks should not prevent the checkin operation from succeeding. I will change the handling by logging a warning in case something goes wrong and proceed with the checking code.

Reply:
Redefined the available rollbacks  as `json.RawMessage` in the openapi model and parse it lazily in the `parseAvailableRollbacks` function, otherwise fleet-server would always respond 400 while it's eagerly unmarshalling the request.
You can find this change in https://github.com/elastic/fleet-server/pull/5975/commits/01f125e3a02cbce7649ad0d01973e04279e16368

I am reusing the model.Available rollback since at this point request and model on ES have the same structure but I am reluctant about doing a complete passthrough from `json.RawMessage` to `json.RawMessage` (as it's currently done for components for example) since that would tightly couple what the agent produces as payload to what is supported/read/used by Fleet UI from elasticsearch.

Reply:
Moved the rollbacks under `upgrade.rollbacks` key in checkin body with commit f9a6fa452ca977a8cba3f33fbdb32eaf69a4ef61 and agent PR https://github.com/elastic/elastic-agent/pull/11753
tested e2e and it works correctly